### PR TITLE
feat: Add support for the event `GUILD_AUDIT_LOG_ENTRY_CREATE`

### PR DESCRIPTION
**Pull request by GeomKid**
## About

This pull request is about adding the Guild Audit Log Entry Event
## Checklist

- [x] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [x] I've ensured the change(s) work on `3.8.6` and higher.
- [ ] I have added the `versionadded`, `versionchanged` and `deprecated` to any new or changed user-facing function I committed.

### Pull-Request specification

I've made this pull request: (check all that apply)
  - [x] For the documentation
  - [x] To add a new feature
  - [ ] As a general enhancement
  - [ ] As a refactor of the library/the library's code
  - [ ] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [ ] A breaking change

### CODE REVIEW

**Review by Damego — CHANGES_REQUESTED** [review]
> The inner payload is an Audit Log Entry object with an additional `guild_id` key

**Review by Damego — CHANGES_REQUESTED** [review]
example of the event payload
```py
{
 'action_type': 25,
 'changes': [{'key': '$add',
              'new_value': [{'id': '855471113685172224',
                             'name': 'new role'}]}],
 'guild_id': '829333896561819648',
 'id': '1063731503076491275',
 'target_id': '972871411741888572',
 'user_id': '143773579320754177'
}
```

**Comment by Damego** [comment]
You could just subclass `AuditLogEntry`

**Comment by GeomKid** [comment]
I'll just do a subclass

File: interactions/api/models/gw.py [hunk 1]
```diff
@@ -206,6 +208,17 @@ class EmbeddedActivity(DictSerializerMixin):
     channel_id: Snowflake = field(converter=Snowflake)
 
 
+@define()
+class GuildAuditLogEntry(AuditLogEntry):
+    """
+    A class object representing an GuildAuditLogEntry.
+    A derivation of AuditLogEntry.
+    :ivar Snowflake guild_id: The guild ID of event.
+    """
```
Comment (Damego) [inline interactions/api/models/gw.py]:
```suggestion
    """
    .. versionadded:: 4.4.0

    A class object representing an GuildAuditLogEntry.
    A derivation of AuditLogEntry.

    :ivar Snowflake guild_id: The guild ID of event.
    """
```

File: interactions/api/models/gw.py [hunk 2]
```diff
@@ -206,6 +208,20 @@ class EmbeddedActivity(DictSerializerMixin):
     channel_id: Snowflake = field(converter=Snowflake)
 
 
+@define()
+class GuildAuditLogEntry(AuditLogEntry):
+    """
+    .. versionadded:: 4.4.0
+
+    A class object representing an GuildAuditLogEntry.
```
Comment (Damego) [inline interactions/api/models/gw.py]:
```suggestion
    A class object representing the event ``GUILD_AUDIT_LOG_ENTRY_CREATE``.
```

**Review by Damego — APPROVED** [review]

**Pull request merged**